great_expectations/datasource/fluent/pandas_datasource.py

```diff
@@ -338,9 +338,7 @@ def _short_id() -> str:
 class DataFrameAsset(_PandasDataAsset, Generic[_PandasDataFrameT]):
     # instance attributes
     type: Literal["dataframe"] = "dataframe"
-    dataframe: Optional[_PandasDataFrameT] = pydantic.Field(
-        default=None, exclude=True, repr=False
-    )
+    dataframe: Optional[_PandasDataFrameT] = pydantic.Field(exclude=True, repr=False)
```

Comment:
@Kilo59 Because when creating `DataFrameAsset` in the new way, we do not supply the `DataFrame`; hence, if there is a default value, we get a type violation in other APIs -- so `DataFrame` here is truly optional: as in, it can be omitted entirely.  Thanks.

Reply:
@Kilo59 If that is the case, how come it works right now (i.e., without the default value)?  In fact, with the default value, I had run into type checking problems in other methods.  Please advise for the best path forward.  Thanks.

Reply:
The fix has been committed -- along with `depracted_api` and `new_api` decorators with messages.